test/suites-ostree/block-upgrade-2-minor.robot

```diff
@@ -0,0 +1,61 @@
+*** Settings ***
+Documentation       Tests related to upgrading MicroShift
+
+Resource            ../resources/common.resource
+Resource            ../resources/ostree.resource
+Library             Collections
+
+Suite Setup         Setup
+Suite Teardown      Teardown
+
+Test Tags           ostree
+
+
+*** Variables ***
+${USHIFT_HOST}                  ${EMPTY}
+${USHIFT_USER}                  ${EMPTY}
+
+${TOO_NEW_MICROSHIFT_REF}       ${EMPTY}
+
+
+*** Test Cases ***
+Upgrading MicroShift By Two Minor Versions Is Blocked
+    [Documentation]    Test verifies if attempt to upgrade MicroShift from
+    ...    X.Y-1 to X.Y+1 is blocked (skipping current X.Y).
+
+    Wait For Healthy System
+    ${initial_deploy_backup}=    Get Future Backup Name For Current Boot
+
+    Deploy Commit Expecting A Rollback    ${TOO_NEW_MICROSHIFT_REF}
+
+    Wait For Healthy System
+    Backup Should Exist    ${initial_deploy_backup}
+    Journal Should Have Information About Failed Version Comparison
```

Comment:
I'm originally assumed we would reuse the existing failed upgrade suite. For test code I do lean towards a bit of repetition for clarity, so it's not a problem. How different are the 2 files aside from this keyword?

Reply:
`failed-upgrade.robot` uses test-agent to block the backup to check if data will be backed up after rollback.
This doesn't add anything extra beside providing two commits too far apart

Reply:
I was thinking we should be able to have 1 test suite file that says "starting from image A and rebasing to image B, the upgrade will fail and that failure is detected and handled correctly". Then the _reason_ for the failure is built into image B.

Maybe that's too simplistic, and we need the tests to configure image B to ensure that failure, like setting the agent sometimes or a version number other times? If that's the case, perhaps we could at least reuse a lot of the keywords, like you're doing here.

I don't think we should hold anything up over this, but it's something to think about as we evolve over time.

Reply:
This sounds very good, though "the reason for the failure is built into image B" is kind of problematic given how much time is needed to build a commit.

Is three-phase build correct assumption? Like `rhel92~microshift-src~fail-greenboot.toml`
Rebuilding RPMs will kick off builds of phase two and three. Seems like a choice between dev productivity and conciseness (not duplicating the test cases with just a minor difference).

Maybe we could use a test suite variable provided by the scenario.sh that would influence test agent set up. Idk, just some random thoughts